coursier.bzl

```diff
@@ -763,8 +763,17 @@ def make_coursier_dep_tree(
     # Behavior: If root module A excludes module X, but root module B requires X, module X will still be fetched.
     artifact_coordinates = []
     exclusion_lines = []
+    forced_versions = []
     for a in artifacts:
-        artifact_coordinates.append(utils.artifact_coordinate(a))
+        coordinates = utils.artifact_coordinate(a)
+
+        # Special-case handling of specific versions that are requested. Without this, coursier will claim that
+        # if a lower version of a dependency is requested by a transitive dep than what is specified here, a
+        # version mismatch will occur, and this will fail the resolution.
+        if a.get("force_version", False):
+            forced_versions.append(coordinates)
```

Comment:
What happens with multiple `maven.artifact` with the same `group:artifact` coordinates enabling `force_version`?

Reply:
I've no idea. I don't expect people will use this much, so I'm okay with this being undefined behaviour until the first bug report comes in and tells us what people expect.